File: py/year2022/day01.py
with open(f"input/year2022/day01.txt", "r") as f:
    lines = f.read().splitlines()

elves = []
elf = []
for line in lines:
    if line == "":
        elves.append(elf)
        elf = []
    else:
        elf.append(int(line))

elves = sorted([sum(elf) for elf in elves])

print(elves[-1])
print(sum(elves[-3:]))

File: rust/src/bin/year2022/day01.rs
use std::fs;

pub fn run() {
    let mut text = fs::read_to_string("../input/year2022/day01.txt").unwrap();
    text = text.replace("\r\n", "\n");
    let elves = text.split("\n\n");

    let mut ration_amounts: Vec<i32> = elves
        .map(|elf| {
            elf.split("\n")
                .map(|ration| match ration.parse::<i32>() {
                    Ok(n) => n,
                    Err(_e) => 0,
                })
                .sum::<i32>()
        })
        .collect::<Vec<i32>>();

    ration_amounts.sort();

    println!("{}", ration_amounts.last().unwrap());
    println!("{}", ration_amounts.iter().rev().take(3).sum::<i32>());
}
